pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -669,6 +669,21 @@ func (p *staticPolicy) GetPodTopologyHints(logger logr.Logger, s state.State, po
 	}
 }
 
+// candidateNUMANodesForHints returns the NUMA IDs that should be considered for hint generation.
+// If memory information is unavailable, it falls back to all NUMA nodes that host CPUs.
+func (p *staticPolicy) candidateNUMANodesForHints() cpuset.CPUSet {
+	numaNodes := p.topology.CPUDetails.NUMANodes()
+	if p.topology.NUMANodesWithMemory.Size() == 0 {
+		return numaNodes
+	}
+
+	numaNodesWithMemory := numaNodes.Intersection(p.topology.NUMANodesWithMemory)
+	if numaNodesWithMemory.Size() == 0 {
+		return numaNodes
+	}
+	return numaNodesWithMemory
```

Comment:
Is there ever a case where `numaNodesWithMemory != numaNodes`? If there is, then let's continue the discussion on how this commit could be simplified. If there's not, I'm not convinced we need any changes to the `cpumanager` at all. My understanding is that we would never have a NUMA node associated with a CPU if it didn't also have memory attached to it.